devito/finite_differences/derivative.py

```diff
@@ -141,6 +149,24 @@ def __new__(cls, expr, *dims, **kwargs):
         obj._x0 = kwargs.get('x0', None)
         return obj
 
+    def __call__(self, x0=None, fd_order=None, side=None):
+        self._fd_order = fd_order or self._fd_order
```

Comment:
Actually, just noticed this. State can't and shouldn't change. And in fact, i don't see why you need to change it at this point

Reply:
This is changed here because the rest is ignored if x0 is None.